chain/cosmos/chain_node.go

```diff
@@ -406,6 +406,7 @@ func (tn *ChainNode) TxCommand(keyName string, command ...string) []string {
 		"--from", keyName,
 		"--gas-prices", tn.Chain.Config().GasPrices,
 		"--gas-adjustment", fmt.Sprint(tn.Chain.Config().GasAdjustment),
+		"--gas", "auto",
```

Comment:
I think we should avoid adding this flag at this level.


I'm not sure if you caught these changes but we (somewhat) recently added the ability to pass extra args to the `InstantiateContract` function. So you should be able to pass this `"--gas", "auto"` flag there.

Is this where you are needing this flag?

Reply:
I missed that, I believe it was why I had this flag. It may affect my local-interchain impl so I will need to test more. b8ef5ef720abbcf3a45bd7b8bcbe1f509e2e78a5